docs/data/sdks/go.md

```diff
@@ -0,0 +1,405 @@
+		UseBatch:        true,
+		OptOut:          false,
+		ServerURL:       "proxy url that forwarding the requests",
+	}
+```
+
+| <div class="big-column">Name</div> | Description  |
+| --- | --- |
+| `APIKey` | Required. string. The API key of the Amplitude project. Events sent by the client struct can be found under this project. Set when you initialize the client struct. |
+| `FlushQueueSize` | int. Events wait in the buffer and are sent in a batch. The buffer is flushed when the number of events reaches `FlushQueueSize`. Defaults to 200.|
+| `FlushInterval` | time.Duration. Events wait in the buffer and are sent in a batch. The buffer is flushed every `FlushInterval`. Defaults to 10 seconds.|
+| `FlushMaxRetries` | int. The number of times the client retries an event when the request returns an error. Defaults to 12 |
+| `Logger` | Logger interface. The logger used by Amplitude client. Defaults to using a wrapper of [Go standard Logger](https://pkg.go.dev/log#Logger): `log.Logger`. |
+| `MinIDLength` | int. The minimum length of `UserID` and `DeviceID`. Defaults to 5.|
+| `Callback`  | func. Optional. Client level callback function. Takes three parameters:<br> 1. event: an Event struct<br> 2. code: a integer of HTTP response code <br> 3. message: a string message. Defaults to `None`. |
+| `ServerZone` |string. The server zone of the projects. Supports `EU` and `US`. For EU data residency, Change to `EU`. Defaults to `US`. |
+| `UseBatch` | bool. Uses HTTP v2 API endpoint if set to `false`, otherwise use batch API endpoint. More about difference between HTTP v2 and batch can be found [here](https://developers.amplitude.com/docs/batch-event-upload-api). Defaults to `false` |
+| `ServerURL` | string. The API endpoint URL that events are sent to. Automatically selected by `ServerZone` and `UseBatch`. If this field is set with a string value instead of `nil`, then `ServerZone` and `UseBatch` are ignored and the string value is used. Defaults to the HTTP API V2 endpoint. |
+| `Storage` | Storage interface. Used to create storage struct to hold events in the storage buffer. Events in storage buffer are waiting to be sent. Defaults to `InMemoryStorage`. |
+| `OptOut`  | bool. Opt out option. If set to `true`, client doesn't process and send events. Defaults to `false`. |
+
+### Track an event
+
+Events represent how users interact with your application. For example, “Button Clicked” may be an action you want to note.
+
+```Go
+
+// Track a basic event
+// One of UserID and DeviceID is required
+event := amplitude.Event{
+    EventOptions: amplitude.EventOptions{UserID: "user-id"},
+    EventType:    "Button Clicked",
+}
+client.Track(event)
+
+// Track events with optional properties
+client.Track(amplitude.Event{
+		EventType:       "type-of-event",
+		EventOptions:    amplitude.EventOptions{
+			UserID:             "user-id",
+			DeviceID:           "device-id",
+		},
+		EventProperties: map[string]interface{}{"source": "notification"},
```

Comment:
Any chance we dont need to add `map[string]interface{}`? or have something shorter?

Reply:
It's required to match data types. `EventProperties`'s data type is `map[string]interface{}`. I guess no way to get rid of it